### ISSUE TITLE
Non-fatal errors are grouping up in dashboard [ANDROID]

### ISSUE THREAD

**Issue by proProbe** [corymsmith/react-native-fabric#77]
Non-fatal JS-errors are all grouped together when using logException meaning that its not possible to actually see all the different logs that has been done.

Everything is grouped under:
`Non-fatal Exception: java.lang.RuntimeException`

Is it possible to go around this somehow so the errors are grouped correctly?

**Comment by willmcclellan** [comment]
I'm seeing this as well. Only the latest non-fatal error is visible in Fabric

**Comment by cooperka** [comment]
I'm getting the same issue, it's very frustrating. It does look like you can see each individual non-fatal report with the **View all sessions** button (see screenshot). It's a halfway-decent workaround but still not very good.

![screen shot 2016-11-22 at 16 33 40](https://cloud.githubusercontent.com/assets/2047062/20542880/a3acacfc-b0d1-11e6-890e-2db3a4e3647d.png)

**Comment by proProbe** [comment]
yeah its kinda getting frustrating. Does anyone have an idea how to solve this?

**Comment by cooperka** [comment]
Apparently Fabric really isn't designed to handle JS non-fatals (at least not currently)... I've switched to [BugSnag](https://docs.bugsnag.com/platforms/react-native/) which explicitly supports React Native. So far I like it a lot better, even on just the free tier.

**Comment by proProbe** [comment]
this seems to be fixed? can anyone confirm?

**Comment by ikzjfr0** [comment]
@proProbe yes, i checked that this morning here

**Comment by Gumija** [comment]
I don't think this is fixed. If you use Crashlytics.logException it is always a RuntimeException and so grouped together.

```
    public void logException(String value) {
            Crashlytics.logException(new RuntimeException(value));
        }
```

Source: https://github.com/corymsmith/react-native-fabric/blob/master/android/src/main/java/com/smixx/fabric/SMXCrashlytics.java#L31

**Comment by moret** [comment]
As @Gumija pointed out, using plain `Crashlytics.logException` still groups all errors on the interface under the same Java file and line - i.e., "SMXCrashlytics.java:33".

To work around that I'm using `Crashlytics.recordCustomExceptionName` directly. I followed the example of [react-native-fabric-crashlytics](https://github.com/mikelambert/react-native-fabric-crashlytics/blob/master/index.js#L18) and used [stacktrace-js's StackTrace.fromError](https://www.stacktracejs.com) to prepare the stack and keep column numbers around:
```js
const stackTrace = await StackTrace.fromError(error, { offline: true });
const stackTraceWithFilename = stackTrace.map(trace => (
  {
    fileName: trace.source,
    functionName: trace.functionName,
  }
));
Crashlytics.recordCustomExceptionName(`${isFatal ? 'fatal': 'non-fatal'}:`, message, stackTraceWithFilename);
```
If column numbers are not needed, just delivering the result of `fromError` should be enough.

This is still not ideal, as we can't disable the line break on the exception name and must make an adaptation to keep the column numbers around, but it's an improvement on the default handling.

**Comment by mdstaff** [comment]
@moret I've seen the docs refer to recordCustomExceptionName as a rather expensive method call ("This call can be expensive and should only be used shortly before process termination.").  Have you seen any performance issues after calling this method?

**Comment by moret** [comment]
@mdstaff no, but I actually only call it on terminal exceptions as the docs indicate, so I probably wouldn't notice if it did.
